tests/components/shelly/test_event.py

```diff
@@ -64,6 +64,51 @@ async def test_rpc_button(
     assert state.attributes.get(ATTR_EVENT_TYPE) == "single_push"
 
 
+@pytest.mark.usefixtures("entity_registry_enabled_by_default")
+async def test_rpc_script_event(
+    hass: HomeAssistant,
+    mock_rpc_device: Mock,
+    entity_registry: EntityRegistry,
+    monkeypatch: pytest.MonkeyPatch,
+) -> None:
+    """Test script event."""
+    await init_integration(hass, 2)
+    entity_id = "event.test_script_js"
+
+    state = hass.states.get(entity_id)
```

Comment:
We can use a snapshot here to test the exact state of the entity, like here: https://github.com/home-assistant/core/blob/dev/tests/components/shelly/test_sensor.py#L1412-L1428

Reply:
I've made the `get_rpc_script_event_types` function return a sorted list, because there where run-to-run differences where the order of event_types would change. This resulted in the snapshot test sometimes failing.